apps/arena/lib/arena/game/skill.ex

```diff
@@ -5,17 +5,19 @@ defmodule Arena.Game.Skill do
   alias Arena.Entities
   alias Arena.Game.Player
 
-  def do_mechanic(game_state, player, mechanics, skill_params) when is_list(mechanics) do
+  def do_mechanic(game_state, entity, player, mechanics, skill_params) when is_list(mechanics) do
```

Comment:
Yeah! i think the one parameter that's not necessary is `player` but since we're using it for some things i had to keep it. With your comment i think i can get to something where player is not needed as parameter, thanks! also really good comments